Looping

AndrewRigg opened the pull request:
This is finally sorted.  (Mostly I think)
Have a play around and tell me if you like it, any suggestions for improvements and any bugs you find.

Golpette commented:
Looks great man, good effort on that. I did one wordsearch but didn't have time to try and break it - will do so over weekend.

(1) what happened to "Show solution"?
(2) Maybe add a wee congratulations once all the words have been found?

Happy for you to merge it when you're happy

Also, one more thing I learned today:  when I pulled this branch it automatically merged with the one I was working on... I expected it create a new branch. So basically, I don't even understand the "pull" command after all this time lol

AndrewRigg commented:
Fixed the problem with the solution button.  It just turns the opacity of the labels on and of with the button press.
Also added congratulations and enough layers so there should never be problems with looping. (8 so that if it was captured as an end point from all 8 directions, it would still be fine.  (Anything more shouldn't be allowed by the algorithm due to overlapping letters etc).

Check it out and merge if happy!

AndrewRigg commented:
Just noticed a problem.  If you go over the same word, it draws the loops on different layers, thereby removing a usable layer for another word.  (Need to allow each correct word to be looped only once each)
